src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,163 @@
+            return parseUnmarkCommand(fullCommand);
+        }
+        else if (command.equals("bye")) {
+            return new ExitCommand();
+        }
+        else {
+            throw new DukeException("I don't recognize that command.");
+        }
+    }
+
+    /**
+     * Parses add deadline command.
+     *
+     * @param fullCommand String to parse.
+     * @return AddCommand instance.
+     */
+    public Command parseAddDeadlineCommand(String fullCommand) {
+        String dataString = fullCommand.replaceFirst("deadline ", "");
+        String[] data = dataString.split("/by ");
+        Task task = new Deadline(data[0], LocalDate.parse(data[1]));
+        return new AddCommand(task);
+    }
+
+    /**
+     * Parses add event command.
+     *
+     * @param fullCommand String to parse.
+     * @return AddCommand instance.
+     */
+    public Command parseAddEventCommand(String fullCommand) {
+        String dataString = fullCommand.replaceFirst("event ", "");
+        String[] data = dataString.split("/at ");
+        Task task = new Event(data[0], LocalDate.parse(data[1]));
+        return new AddCommand(task);
+    }
+
+    /**
+     * Parses add todo command.
+     *
+     * @param fullCommand String to parse.
+     * @throws DukeException If command is not recognized.
+     * @return AddCommand instance.
+     */
+    public Command parseAddTodoCommand(String fullCommand) throws DukeException {
+        String dataString = fullCommand.replaceFirst("todo ", "");
+        if (dataString.trim().equals("")) {
+            throw new DukeException("Todo needs a description");
+        }
+        Task task = new Todo(dataString);
+        return new AddCommand(task);
+    }
+
+    /**
+     * Parses list command.
+     *
+     * @param fullCommand String to parse.
+     * @return ListCommand instance.
+     */
+    public Command parseListCommand() {
+        return new ListCommand();
+    }
+
+    /**
+     * Parses delete command.
+     *
+     * @param fullCommand String to parse.
+     * @return DeleteCommand instance.
+     */
+    public Command parseDeleteCommand(String fullCommand) {
+        String dataString = fullCommand.replaceFirst("delete ", "");
+        int taskNumber = Integer.parseInt(dataString);
+        return new DeleteCommand(taskNumber);
+    }
+
+    /**
+     * Parses mark command.
+     *
+     * @param fullCommand String to parse.
+     * @return MarkCommand instance.
+     */
+    public Command parseMarkCommand(String fullCommand) {
+        String dataString = fullCommand.replaceFirst("mark ", "");
+        int taskNumber = Integer.parseInt(dataString);
+        return new MarkCommand(taskNumber);
+    }
+
+    /**
+     * Parses unmark command.
+     *
+     * @param fullCommand String to parse.
+     * @return UnmarkCommand instance.
+     */
+    public Command parseUnmarkCommand(String fullCommand) {
+        String dataString = fullCommand.replaceFirst("unmark ", "");
```

Comment:
Commendable discretion shown here to differentiate when to use camelCase and when to use PascalCase.